### PR TITLE
docs: add sudo for install commands

### PR DESCRIPTION
**Pull request by caicancai**
## Which issue does this PR close?

docs: add sudo for install commands

## Rationale for this change

For linux, add sudo is necessary

### CODE REVIEW

**Review by alamb — APPROVED** [review]
Thank you @caicancai

File: docs/source/contributor-guide/index.md [hunk 1]
```diff
@@ -95,7 +95,7 @@ Compiling DataFusion from sources requires an installed version of the protobuf
 On most platforms this can be installed from your system's package manager
 
 ```
-$ apt install -y protobuf-compiler
+$ sudo apt install -y protobuf-compiler
 $ dnf install -y protobuf-devel
```
Comment (alamb) [inline docs/source/contributor-guide/index.md:99]: I wonder if we should add `sudo` to the other similar commands (like `dnf` and `pacman`) 🤔

Comment (caicancai) [reply docs/source/contributor-guide/index.md:99]: I have tested it on my Linux machine. Pacman and dnf can be used without sudo.
But apt install -y protobuf-compiler seems to require sudo under ordinary users.

Comment (caicancai) [reply docs/source/contributor-guide/index.md:99]: But the dnf command also seems to require apt installation. Do you need to add something?

**Review by Weijun-H — APPROVED** [review]
LGTM! Thanks @caicancai

File: docs/source/contributor-guide/index.md (hunk 1 above)
Comment (alamb) [reply docs/source/contributor-guide/index.md:99]: Let's start with this change and we can always update the instructions later if we find something that could make it clearer

Thanks again @caicancai

**Review by alamb — APPROVED** [review]

**Pull request merged**